resolver/src/main/java/io/netty/resolver/RoundRobinInetAddressResolver.java

```diff
@@ -97,4 +97,17 @@ public void operationComplete(Future<List<InetAddress>> future) throws Exception
             }
         });
     }
+
+    /**
+     * @param anyInt any integer value to covert
+     * @param len max limit of range
+     * @return integer in (0, len] range, using formula {@code (anyInt % len + anyInt) % len}
+     */
+    protected static int toIndex(int anyInt, int len) {
+        int index = anyInt % len;
+        if (index < 0) {
+            index = index + len;
```

Comment:
After some experiences by overflowing:

```
  protected static int toIndex1(int anyInt, int len) {
    int index = anyInt % len;
    if (anyInt < 0) {
      index = index + len - 1;
    }
    return index;
  }

  protected static int toIndex2(int anyInt, int len) {
    int index = anyInt % len;
    if (index < 0) {
      index = index + len;
    }
    return index;
  }

  public static void main(String[] args) {
    int index = Integer.MAX_VALUE - 5;
    for (int i = 0; i < 10; i++) {
      System.out.println("With index1 function for index = " + index + ", index = " + toIndex1(index, 3));
      System.out.println("With index2 function for index = " + index + ", index = " + toIndex2(index, 3));
      System.out.println();
      index++;
    }
}
```

I have come to the conclusion that both formulas are right, but yours will do less computation in cases where it is divisible `when anyInt % len == 0` so I guess I prefer yours ;-)